config/config.go

```diff
@@ -547,7 +584,7 @@ func InitServer(sType ServerType) error {
 	// they have overridden the defaults.
 	hostname = viper.GetString("Server.Hostname")
 
-	if sType.IsSet(CacheType) {
+	if IsServerEnabled(CacheType) {
```

Comment:
This should be a current server check. If you are running both an origin and a cache from the same process and you start up the cache first, then this will overwrite the origin's xrootd port, which we don't want.